zone/bot_command.cpp

```diff
@@ -2216,24 +2362,47 @@ namespace ActionableBots
 		return ab_type;
 	}
 
-	// Returns single, scoped bot
 	static Bot* AsGroupMember_ByClass(Client *bot_owner, Client *bot_grouped_player, uint8 cls, bool petless = false) {
 		if (!bot_owner || !bot_grouped_player)
 			return nullptr;
-		if (!bot_grouped_player->GetGroup())
+		if (!bot_grouped_player->GetGroup() && !bot_grouped_player->GetRaid())
 			return nullptr;
 
-		std::list<Mob*> group_list;
-		bot_grouped_player->GetGroup()->GetMemberList(group_list);
-		for (auto member_iter : group_list) {
-			if (!MyBots::IsMyBot(bot_owner, member_iter))
-				continue;
-			if (member_iter->GetClass() != cls)
-				continue;
-			if (petless && member_iter->GetPet())
-				continue;
 
-			return static_cast<Bot*>(member_iter);
+		if (bot_owner->IsRaidGrouped()) {
+			Raid* raid = bot_grouped_player->GetRaid();
+			if (!raid)
+				return nullptr;
+			uint32 raid_group = raid->GetGroup(bot_grouped_player);
+			if (raid_group >= MAX_RAID_GROUPS) {
+				return nullptr;
+			}
+			for (int x = 0; x < MAX_RAID_MEMBERS; x++) {
```

Comment:
Can probably do
```cpp
for (const auto& m : raid->members) {
    if (m.member) {
```

Don't think we have to check if mob is in zone if we have a valid pointer.